enhancements/plain-k8s-bundles.md

```diff
@@ -0,0 +1,220 @@
+
+The following are parts of a CSV that are NOT directly inferrable from plain kubernetes resources:
+
+- Name
+- Descriptors
+- Description
+- Display Name
+- Keywords
+- Labels
+- Links
+- Maintainers
+- Maturity
+- Provider
+- Selector
+- Icon
+- Runtime Metadata
+- MinKubeVersion
+- (Future) Optional APIs
+- Replaces
+- Version
+- InstallModes
+
+This non-inferrable data will be gathered through a mixture of resource annotations and an `olm.yaml` file in the `metadata` directory of the bundle.
+
+Ex. _olm.yaml_
+
+```yaml
+name: packageserver-v1.14.1
+maturity: alpha
+version: 0.14.1
+replaces: packageserver-v1.14.0
+displayName: Package Server
+description: Represents an Operator package that is available from a given CatalogSource which will resolve to a ClusterServiceVersion.
+minKubeVersion: 1.11.0
+keywords: ['packagemanifests', 'olm', 'packages']
+maintainers:
+- name: Red Hat
+email: openshift-operators@redhat.com
+provider:
+name: Red Hat
+links:
+- name: Package Server
+url: https://github.com/operator-framework/operator-lifecycle-manager/tree/master/pkg/package-server
+installModes:
+- type: OwnNamespace
+  supported: true
+- type: SingleNamespace
+  supported: true
+- type: MultiNamespace
+  supported: true
+- type: AllNamespaces
+  supported: true
+icon: my-icon-png-base64-encoded
+labels:
+  operated-by: packageserver
+```
+
+#### `opm` Support
+
+[`opm`](https://github.com/operator-framework/operator-registry#overview) must be updated to provide support for building `k8s+v1` bundles. This can be acheived with explicit flags or through CSV/olm.yaml file presence.
+
+### CSV Synthesis
+
+OLM will support `k8s+v1` bundles by synthesizing an CSV from its contents. Resources usually generated by a CSV (e.g. APIService and Service) will not be directly applied, but instead be projected onto a synthesized CSV.
+
+This intermediary step (creating a CSV internally to represent the kubernetes objects) is a delibrate decision in 4.5 to help focus and deliver the feature. This may be relaxed in future release versions.
+
+### Indexes
+
+When a `k8s+v1` bundle is added to an index, `opm` will also synthesize and add a CSV. A lot of the existing `opm` and `operator-registry` code assumes bundles always contain CSVs, which make synthesis the simplest path to supporting `k8s+v1` bundles -- in the future, we may want to revisit this decision, especially when the [`Operator` API](https://github.com/openshift/enhancements/blob/master/enhancements/olm/simplify-apis.md#proposal) is released.
+
+To fill out the dependency graph between versions of bundles, OLM considers the "replaces" field on the CSV which points to the previous version of the operator. This field will need to be supplied in the additional metadata the operator author provides in the bundle so that OLM can build dependency graphs. There is another enhancement around semantic versioning that would remove this requirement altogether, but it is independent of this enhancement.
```

Comment:
Could we sidestep this entirely by just assuming that a version is required and `--mode=semver` is assumed?

Reply:
I think versions are required for current index builds anyway. I'll call out the semver enhancement as a prerequisite somewhere.